src/git-prune-branches

```diff
@@ -7,59 +7,21 @@ function preconditions {
 }
 
 
-function should_delete_branch {
-  local branch_name=$1
-
-  if [ "$(is_feature_branch "$branch_name")" = true ]; then
-    if [ "$INITIAL_BRANCH_NAME" = "$branch_name" ] && [ "$INITIAL_OPEN_CHANGES" = true ]; then
-      echo false
-    else
-      echo true
-    fi
-  else
-    echo false
-  fi
-}
-
-
 function steps {
-  local initial_branch_deleted=false
-
-  if [ "$INITIAL_BRANCH_NAME" != "$MAIN_BRANCH_NAME" ]; then
-    echo_if_true "stash_open_changes" "$INITIAL_OPEN_CHANGES"
-    echo "checkout_main_branch"
-  fi
+  for branch_to_delete in $(local_branches_with_gone_remote); do
 
-  for local_branch_name in $(local_merged_branches); do
-    if [ "$(should_delete_branch "$local_branch_name")" = true ]; then
-      if [ "$INITIAL_BRANCH_NAME" = "$local_branch_name" ]; then
-        initial_branch_deleted=true
-      fi
-      if [ "$(has_tracking_branch "$local_branch_name")" = true ]; then
-        echo "delete_remote_branch $local_branch_name"
-      fi
-      echo "delete_local_branch $local_branch_name"
+    # check out the main branch if we are deleting the branch we are currently on
+    if [ "$INITIAL_BRANCH_NAME" = "$branch_to_delete" ]; then
+      echo "checkout_branch $MAIN_BRANCH_NAME"
     fi
-    echo_if_true "delete_parent_entry $local_branch_name" "$(knows_parent_branch "$local_branch_name")"
-  done
 
-  for remote_branch_name in $(remote_only_merged_branches); do
-    if [ "$(should_delete_branch "$remote_branch_name")" = true ]; then
-      echo "delete_remote_only_branch $remote_branch_name"
-      echo_if_true "delete_parent_entry $remote_branch_name" "$(knows_parent_branch "$remote_branch_name")"
-    fi
-  done
+    # remove the branch from the branch hierarchy configuration
+    parent_branch_name=$(parent_branch "$branch_to_delete")
+    echo_update_child_branches "$branch_to_delete" "$parent_branch_name"
+    echo_if_true "delete_parent_entry $branch_to_delete" "$(knows_parent_branch "$branch_to_delete")"
```

Comment:
Seems odd that you update the child branches no matter what. What do you think about
```sh
if [ "$(knows_parent_branch "$branch_to_delete")" = true ]; then  
  local parent_branch_name=$(parent_branch "$branch_to_delete")
  echo_update_child_branches "$branch_to_delete" "$parent_branch_name"
  echo "delete_parent_entry $branch_to_delete"
fi
```